Switch to image/next.js

alshakerM opened the pull request:
Switched to image/next.js.

Fixes #86

alshakero commented on components/pages/Messages/Messages.js:
```diff
@@ -188,11 +195,15 @@ export function Messages({ fromUserId }) {
-                          />
+                          threadData.from_user_id && (
+                            <Image
+                              src={threadData?.from_user_thumbnail}
```
We know threadData is not null here.
```suggestion
                              src={threadData.from_user_thumbnail}
```

alshakero commented on components/pages/Messages/Messages.module.css:
```diff
@@ -62,8 +62,6 @@
   background-color: #fafafa;
 }
 .senderThumbnail {
-  height: 56px;
-  width: 56px;
```
These should stay.

alshakero commented on components/pages/Messages/Messages.module.css:
```diff
@@ -93,8 +91,6 @@
   box-sizing: border-box;
 }
 .messagesSenderAvatar {
-  height: 24px;
-  width: 24px;
```
These too.

alshakero commented on next.config.js:
```diff
@@ -1,12 +1,5 @@
-    ];
+  images: {
+    domains: ['picsum.photos'],
   },
```
Why delete these? They're needed to redirect from /direct to /direct/inbox.

alshakero reviewed (changes requested):
Clean! But for `Image`. The `width` and `height` are the width and height of the jpg file itself, not for how you'd like to appear. So keep sizes in CSS and change `width` and `height` values to the JPG's.

alshakerM commented:
Changed the width and height of Next.js/Image to be the original pic width and height.
Added a parent div to the Next.js/Image to add the needed size for the Image

alshakero commented on components/Post/Post.js:
```diff
@@ -64,7 +63,7 @@ function PostActions({ postId, is_post_liked, isExpanded }) {
         <LikeButton
           is_post_liked={is_post_liked}
-          onClick={() => togglePostLike(postId)}
+          onClick={() => postLike(postId, !is_post_liked)}
```
This is out of scope. Let's leave it this time, but next time please create another tiny PR for such things.

alshakerM replied:
I think this is from the rebase.

alshakero commented on components/Post/Post.js:
```diff
@@ -621,7 +626,7 @@ export function Post({ isIndependentPost, datum, index, isFloating }) {
 
           <PostDate posting_time={datum.posting_time} isExpanded={isExpanded} />
-          <PostInput index={index} isExpanded={isExpanded} />
+          <PostInput index={datum.post_id} isExpanded={isExpanded} />
```
Same.

alshakerM replied:
+ 1

alshakero commented on components/PostImage/PostImage.js:
```diff
@@ -1,6 +1,29 @@
 import styles from './PostImage.module.css';
 import cx from 'classnames';
 import React from 'react';
+import Image from 'next/image';
+
+const useElementOnScreen = (options) => {
+  const imgRef = React.useRef(null);
+  const [isVisible, setIsVisible] = React.useState(false);
+  React.useEffect(() => {
+    const observer = new IntersectionObserver(([entry]) => {
+      if (entry.isIntersecting !== isVisible) {
+        setIsVisible(entry.isIntersecting);
+      }
+    }, options);
+    if (imgRef.current) {
+      observer.observe(imgRef.current);
+    }
+    return () => {
+      if (imgRef.current && isVisible) {
+        observer.disconnect(imgRef.current);
+      }
+    };
+  }, [imgRef, options]);
+
+  return [imgRef, isVisible];
+};
```
This doesn't belong here 🤔

alshakerM replied:
Addressed 1 to 1.

alshakero commented on components/PostImage/PostImage.js:
```diff
@@ -9,8 +32,23 @@ export function PostImage({
+  isActive,
 }) {
   const [isImgDoubleClicked, setIsImgDoubleCLicked] = React.useState(false);
+  const [isImgClicked, setIsImgCLicked] = React.useState(false);
+  const [imgRef, isVisible] = useElementOnScreen({
+    root: null,
+    rootMargin: '0px',
+    threshold: 0,
+  });
+
+  React.useEffect(() => {
+    if (!isActive) {
+      setIsImgCLicked(false);
+    }
+  }, [isActive]);
```
Why are these here? :\

alshakerM replied:
Addressed 1 to 1.

alshakero commented on components/PostImage/PostImage.js:
```diff
@@ -50,6 +92,58 @@ export function PostImage({
+              >
+                <div className={styles.taggedContent}>
+                  <div className={styles.emptyArrow}>
+                    <div className={styles.arrow}></div>
+                  </div>
+                  <p className={styles.taggedUserName}>
+                    <a
+                      className={styles.taggedUserLink}
+                      href={`/${tag.tagged_user_name}`}
+                    >
+                      {tag.tagged_user_name}
+                    </a>
+                  </p>
+                </div>
+              </div>
+            )}
+          </>
+        );
+      })}
+      {tags && (
+        <div
+          className={cx(styles.tagIconContainer, {
+            [styles.isVisible]: isVisible,
+            [styles.isClicked]: isImgClicked,
+          })}
+        >
+          <svg
+            aria-label="Tags"
+            className={styles.tagIcon}
+            color="#ffffff"
+            fill="#ffffff"
+            height="12"
+            role="img"
+            viewBox="0 0 24 24"
+            width="12"
+          >
+            <path d="M21.334 23H2.666a1 1 0 01-1-1v-1.354a6.279 6.279 0 016.272-6.272h8.124a6.279 6.279 0 016.271 6.271V22a1 1 0 01-1 1zM12 13.269a6 6 0 116-6 6.007 6.007 0 01-6 6z"></path>
+          </svg>
+        </div>
+      )}
```
The whole tags PR seems here.

alshakerM replied:
Addressed 1 to 1.

alshakero commented on components/pages/UserProfile/UserProfile.module.css:
```diff
@@ -162,9 +162,10 @@
 .postsSection {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  grid-template-columns: 293px 293px 293px;
```
You can change it back to flex so it works on mobiles.

alshakerM replied:
Done.

alshakero commented on mocking-scripts/add-tags-to-media.js:
```diff
@@ -0,0 +1,31 @@
+const { writeFileSync } = require('fs');
+let posts = require('../server/posts-data.json');
+let users = require('../server/profile-data.json');
+
+function getRandomUser() {
+  return users[Math.floor(Math.random() * (users.length - 1))];
+}
+
+posts = posts.map((post, index) => {
+  if (index % 3 === 0) {
+    post.media_items
+      .filter((item) => item.type === 'photo')
+      .map((photo) => {
+        photo.tags = [
+          {
+            coordinates: {
+              x: 200 + Math.floor(Math.random() * 400),
+              y: 200 + Math.floor(Math.random() * 400),
+            },
+            tagged_user_name: getRandomUser().user_name,
+          },
+        ];
+        return photo;
+      });
+    post.post_has_tags = true;
+  }
+  post.post_has_tags = false;
+  return post;
+});
+
+writeFileSync('../server/posts-data.json', JSON.stringify(posts));
```
Tags PR.

alshakerM replied:
Addressed 1 to 1.

alshakero commented on pages/api/posts.js:
```diff
@@ -1,25 +1,113 @@
+  if (req.method === 'POST') {
+    const { action } = req.query;
+    const { postId, like, commentId, text, replyId } = req.body;
+    console.log('omar1', { postId, like, commentId, text, replyId });
```
```suggestion
```

alshakerM replied:
Addressed 1 to 1.

alshakero commented on pages/api/posts.js:
```diff
@@ -1,25 +1,113 @@
+            }`,
+          });
+        } catch (e) {
+          console.log('omar', e);
```
```suggestion

```

alshakerM replied:
Addressed 1 to 1.

alshakero commented on stores/messagesStore/controls.js:
```diff
@@ -13,3 +13,42 @@ export async function REQUEST_THREADS(action) {
 }
+
+export async function REQUEST_MESSAGE_LIKE(action) {
+  await fetch('/api/messages?action=messageLike', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      fromUserId: action.fromUserId,
+      index: action.index,
+      like: action.like,
+    }),
+  });
+}
+
+export async function DELETE_MESSAGE(action) {
+  await fetch('/api/messages?action=deleteMessage', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      fromUserId: action.fromUserId,
+      index: action.index,
+    }),
+  });
+}
+export async function SUBMIT_MESSAGE(action) {
+  await fetch('/api/messages?action=submitMessage', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      fromUserId: action.fromUserId,
+      text: action.text,
+    }),
+  });
+}
```
I can't explain this. I don't know how this happened.

alshakerM replied:
Me either

alshakero reviewed (changes requested):
This PR has the changes of the last 3 PRs together.

alshakero reviewed (approved):
LGTM now. The issue was fixed by a rebase.
